homeassistant/components/ialarmxr/__init__.py

```diff
@@ -0,0 +1,94 @@
+"""iAlarmXR integration."""
+import asyncio
+import logging
+
+from async_timeout import timeout
+from pyialarmxr import IAlarmXR
+
+from homeassistant.components.alarm_control_panel import SCAN_INTERVAL
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    STATE_UNKNOWN,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, IALARMXR_TO_HASS
+
+PLATFORMS = [Platform.ALARM_CONTROL_PANEL]
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up iAlarmXR config."""
+    host = entry.data[CONF_HOST]
+    port = entry.data[CONF_PORT]
+    username = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+
+    ialarmxr = IAlarmXR(host, username, password, port)
+
+    try:
+        async with timeout(10):
+            mac = await hass.async_add_executor_job(ialarmxr.get_mac)
+    except (asyncio.TimeoutError, ConnectionError) as ex:
```

Comment:
We should return False and log an error if authentication fails since we don't want to keep trying a remote api with an incorrect password.  In a future PR you can `raise ConfigEntryAuthFailed` to start a reauth flow.